src/main/java/com/eerussianguy/blazemap/feature/maps/AtlasExporter.java

```diff
@@ -0,0 +1,144 @@
+package com.eerussianguy.blazemap.feature.maps;
+
+import com.eerussianguy.blazemap.api.BlazeRegistry;
+import com.eerussianguy.blazemap.api.maps.Layer;
+import com.eerussianguy.blazemap.api.maps.MapType;
+import com.eerussianguy.blazemap.api.maps.TileResolution;
+import com.eerussianguy.blazemap.api.util.RegionPos;
+import com.eerussianguy.blazemap.engine.BlazeMapAsync;
+import com.eerussianguy.blazemap.engine.StorageAccess;
+import com.eerussianguy.blazemap.engine.client.BlazeMapClientEngine;
+import com.eerussianguy.blazemap.engine.client.LayerRegionTile;
+import com.eerussianguy.blazemap.util.Colors;
+import com.mojang.blaze3d.platform.NativeImage;
+import net.minecraft.client.Minecraft;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.level.Level;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AtlasExporter {
+    private static Task task = null;
+
+    public static synchronized void export(Task task) {
+        if(AtlasExporter.task == null) {
+            AtlasExporter.task = task;
+            BlazeMapAsync.instance().clientChain.runOnDataThread(AtlasExporter::export);
```

Comment:
Nit: From a code confusion standpoint, this function and the one on line 38 should probably be given different names. This line of code confused me for a hot minute as it seems to be pointing at itself! This would be fixed by simply cahnging the name of one or the other.

Reply:
Do you have suggestions?

Reply:
switched to `exportAsync` in working branch

Reply:
Gah, you changed the name in both places, leaving behind the exact same problem you started with 😂

The issue is that, since they're both named the same, it's really confusing what function `AtlasExporter::export` is pointing to. By changing the name of one but not the other, that makes them distinct by more than just their args, making it clear which of the two you're passing to `runOnDataThread` here